### DISCUSSION TITLE
Agent model for subnetwork

### DISCUSSION THREAD

**Discussion by Tchoupinax** [seriousm4x/UpSnap#470] [Feature requests]
Hello,

I would like to be able to wake on lan devices that are not on the same network. Devices are reachable without any issue, the routes does a great job.

However, I learnt ([one source](https://www.reddit.com/r/HomeNetworking/comments/11tj10c/wake_on_lan_on_my_pc_from_outside_my_network/)) that it is not possible to perform a WOL request from another network due to the network layer.

So to do that, I could deploy another instance of `UpSnap` on this network and uses it. But It would be great if all instances could be managed at only one spot. To do that, we would need to have an "agent" deployed on the subnet and the `UpSnap` instance will forward the request to this agent.

It could be just a simple golang program that registers to the instance at start and waits for requests by HTTP.

What do you think?

Have a nice day!

**Comment by seriousm4x** [comment]
https://github.com/seriousm4x/UpSnap/discussions/51
